### ISSUE TITLE
date calculation for different note types needs clarification

### ISSUE THREAD

**Issue by rholloway** [liamcain/obsidian-periodic-notes#132]
I have changed to testing 1.0.0-beta and it seems the {{date}} calculation has changed a little bit. In 0.0.17 you could use {{date:YYYY}} on all note types and it would essentially put the year. In 1.0.0-beta, it seems that the optional YYYY parameter doesn't work and it fails to convert (leaves raw unconverted) - at least if trying to use in something like a quarterly note. Similarly, using {{date}} as is in a quarterly note converts it to 2022-Q3 for instance.

I think the action of using {{date}} output 2022-Q3 in a quarterly note can make sense, but I would expect to be able to use {{date:YYYY}} in quarterly note and have it generate the year based on the README, which states {{date}} is valid for all note types and accepts optional format parameter.

**Comment by xurizaemon** [comment]
I agree, this confused me. There seems to be crossover between this and the behaviour described in https://github.com/liamcain/obsidian-periodic-notes/issues/128 (but #128 is about the wrong filename being created, while this issue is about the `{{date}}` token behaving inconsistently between template types.

As you say, these tokens operate as expected in the daily template. The documentation says that "supported template types" is "all" at https://github.com/liamcain/obsidian-periodic-notes#template-tags but the "optionally accepts a format" does not work consistently between the template types.

Quick visual demo below:

![obsidian-periodic-daily-weekly-date](https://user-images.githubusercontent.com/105608/196004297-8c2742fd-fecc-4b2d-ae4b-539883ba3059.gif)

Using the same input in Daily and Weekly templates:

```yaml
test1: {{date}}
test2: {{date:YYYY-MM-DD}}
test1: {{date:YYYY}}
day: {{date:YYYY-MM-DD}}
week: {{date:GGGG-[W]WW}}
```

The output in a Daily template is:

```
test1: 2022/10/16
test2: 2022-10-16
test1: 2022
day: 2022-10-16
week: 2022-W41
```
and in a Weekly template is:
```
test1: 2022/W43
test2: {{date:YYYY-MM-DD}}
test1: {{date:YYYY}}
day: {{date:YYYY-MM-DD}}
week: {{date:GGGG-[W]WW}}
```

**Comment by jwhitley** [comment]
I’m also seeing this issue. In my case, it breaks the automatic creation of a Dataview query to quickly reference the daily notes associated with that week:

````
# Daily Summary
```dataview
LIST
FROM "50 Resources/Daily Journal"
WHERE typeof(created) = "date" AND created >= date({{date:YYYY-MM-DD}}) AND created < date({{date:YYYY-MM-DD}})+dur(7 day)
SORT created ASC
```
````

The query is included into the template, but the `{{date:YYYY-MM-DD}}` instances are not correctly substituted. In my case, I’m also using a custom formatted date using “Wnn” for the weekly note so I can’t just pass plain `{{date}}` to Dataview.
